src/plugins/okta/login.ts

```diff
@@ -51,7 +59,19 @@ const fetchSsoWebToken = async (
     `https:${org.providerDomain}/oauth2/v1/token`,
     init
   );
-  await validateResponse(response);
+
+  if (!response.ok) {
+    if (response.status === 400) {
+      const data = await response.json();
+      if (data.error === "invalid_grant") {
+        await deleteIdentity();
+        throw new Error("Your Okta session has expired. Please login again.");
+      }
+    }
+
+    await validateResponse(response);
+  }
```

Comment:
A few questions on this block
 
* Are you intentionally not calling `validateResponse` when `response.ok` is true? That seems a weird change
* Are you sure you only want to handle `response.status === 400`? It seems to me what you actually care about is only the `invalid_grant` message
* Are you sure this response must be JSON? `const data = await response.json();` without a try catch around it might be making a huge leap of faith

Reply:
I see, `validateResponse` would more aptly be called `throwIfNotOkResponse`. Sounds good.